roles/smartgateway/templates/deployment.yaml.j2

```diff
@@ -26,58 +26,104 @@ spec:
-        - "60"
+        - "{{ sg_vars.bridge.stats_period }}"
+{%   endif -%}
+{%   if sg_vars.bridge.ring_buffer_count is defined and sg_vars.bridge.ring_buffer_count != "" %}
         - --rbc
-        - "15000"
+        - "{{ sg_vars.bridge.ring_buffer_count }}"
+{%   endif -%}
+{%   if sg_vars.bridge.stop_count is defined and sg_vars.bridge.stop_count != "" %}
         - --count
-        - "{{ data_count }}"
+        - "{{ sg_vars.bridge.stop_count }}"
+{%   endif -%}
+{%   if sg_vars.bridge.inet_target is defined and sg_vars.bridge.inet_target != "" %}
+        - --gw_inet
+        - "{{ sg_vars.bridge.inet_target }}"
+{%   endif -%}
+{%   if sg_vars.bridge.unix_socket_path is defined and sg_vars.bridge.unix_socket_path != "" %}
+        - --gw_unix
+        - "{{ sg_vars.bridge.unix_socket_path }}/smartgateway"
+{%   endif -%}
+{%   if sg_vars.bridge.ring_buffer_size is defined and sg_vars.bridge.ring_buffer_size != "" %}
+        - --rbs
+        - "{{ sg_vars.bridge.ring_buffer_size }}"
+{%   endif -%}
+{%   if sg_vars.bridge.container_id_pattern is defined and sg_vars.bridge.container_id_pattern != "" %}
+        - --cid
+        - "{{ sg_vars.bridge.container_id_pattern }}"
+{%   endif -%}
+{%   if sg_vars.bridge.unix_socket_path is defined and sg_vars.bridge.unix_socket_path != "" %}
         volumeMounts:
-        - mountPath: /tmp
+        - mountPath: {{ sg_vars.bridge.unix_socket_path }}
           name: socket-dir
+{%   endif %}
+{% endif %}
+{% if transports is defined and transports != "" %}
       - name: sg-core
         image: {{ core_container_image_path }}
         args:
           - -config
           - /etc/sg-core/sg-core.conf.yaml
+{%   if sg_vars.bridge.unix_socket_path is defined and sg_vars.bridge.unix_socket_path != "" %}
         volumeMounts:
-        - mountPath: /tmp
+        - mountPath: {{ sg_vars.bridge.unix_socket_path }}
           name: socket-dir
+{%   endif %}
 {%   if smartgateway_resource_configmap is defined %}
         - mountPath: /etc/sg-core/
           name: sg-core-config
           readOnly: true
-{%     if service_type == 'events' %}
+{%   endif %}
+{%   if (applications | selectattr('name','equalto','elasticsearch') | list | count > 0) %}
         - name: elastic-certs
           mountPath: /config/certs
-{%     endif %}
 {%   endif %}
         env:
           - name: MY_POD_NAME
             valueFrom:
               fieldRef:
                 apiVersion: v1
                 fieldPath: metadata.name
+{%   if services is defined and services != "" %}
         ports:
-        - name: "{{ service_type }}"
-          containerPort: 8081
+{%     for service in services %}
+{%       if service.ports is defined and service.ports != "" %}
+{%         for port in service.ports %}
+          - name: {{ port.name }}
+            containerPort: {{ port.target_port }}
+            protocol: {{ port.protocol }}
+{%         endfor %}
+{%       endif %}
+{%     endfor %}
+{%   endif %}
+{% endif %}
```

Comment:
[TODO] documentation note of how ports work. The port name needs to be unique even across multiple services (since services can be a list). This is on a per SG object.

Reply:
Same topic, just fuel for discussion - if you are having to re-document basics about k8s to prevent the user from hanging themselves, maybe they are being offered too much rope. We can talk about the "don't infer" rule offline; not looking for any changes here.